src/execution_plan/execution_plan_reduce_to_apply.c

```diff
@@ -48,60 +46,66 @@ static void _OpBaseSwapChildren(OpBase *op, int a, int b) {
+		anti = true;
+		expression = expression->op.children[0];
+	}
+	// Build new Semi Apply op.
+	OpBase *op_semi_apply = NewSemiApplyOp(plan, anti);
+	const cypher_astnode_t *path = expression->op.children[0]->operand.constant.ptrval;
+	// Add a match branch as a Semi Apply op child.
+	ExecutionPlan_AddOp(op_semi_apply, _buildMatchBranch(plan, path));
+	return op_semi_apply;
+}
+
 /* This method reduces a filter tree into an OpBase. The method perfrom post-order traversal over the
- * filter tree, and checks if if the current subtree rooted at the visited node contains path filter or not,
+ * filter tree, and checks if the current subtree rooted at the visited node contains path filter or not,
  * and either reduces the root or continue traversal and reduction.
  * The three possible operations could be:
  * 1. OpFilter - In the case the subtree originated from the root does not contains any path filters.
  * 2. OpSemiApply - In case of the current root is an expression which contains a path filter.
  * 3. ApplyMultiplexer - In case the current root is an operator (OR or AND) and at least one of its children
  * has been reduced to OpSemiApply or ApplyMultiplexer. */
-static OpBase *_ReduceFilterToOpBase(ExecutionPlan *plan,
-									 FT_FilterNode *filter_root) {
+static OpBase *_ReduceFilterToOp(ExecutionPlan *plan,
+								 FT_FilterNode *filter_root) {
 	FT_FilterNode *node;
 	// Case of an expression which contains path filter.
 	if(filter_root->t == FT_N_EXP && FilterTree_ContainsFunc(filter_root, "path_filter", &node)) {
+		/* If an expression filter tree node contains "path_filter" function, it can be either as
+		 * a single function, or a single child of a single "not" function. ß*/
 		AR_ExpNode *expression = filter_root->exp.exp;
-		bool anti = false;
-		if(strcasecmp(expression->op.func_name, "not") == 0) {
-			anti = true;
-			expression = expression->op.children[0];
-		}
-		// Build new Semi Apply op.
-		OpBase *op_semi_apply = NewSemiApplyOp(plan, anti);
-		const cypher_astnode_t *path = expression->op.children[0]->operand.constant.ptrval;
-		// Add a match branch as a Semi Apply op child.
-		ExecutionPlan_AddOp(op_semi_apply, _buildMatchBranch(plan, path));
-		return op_semi_apply;
+		return _ApplyOpFromPathExpression(plan, expression);
 	}
 	// Case of an operator (Or or And) which its subtree contains path filter
 	if(filter_root->t == FT_N_COND && FilterTree_ContainsFunc(filter_root, "path_filter", &node)) {
 		// Create the relevant LHS branch and set a bounded branch for it.
-		OpBase *lhs = _ReduceFilterToOpBase(plan, filter_root->cond.left);
+		OpBase *lhs = _ReduceFilterToOp(plan, filter_root->cond.left);
 		if(lhs->type == OPType_FILTER) filter_root->cond.left = NULL;
 		_CreateBoundBranch(lhs, plan);
 		// Create the relevant RHS branch and set a bounded branch for it.
-		OpBase *rhs = _ReduceFilterToOpBase(plan, filter_root->cond.right);
+		OpBase *rhs = _ReduceFilterToOp(plan, filter_root->cond.right);
```

Comment:
Care to verify that at least one of the branches (lhs, rhs) isn't OPType_FILTER.

Reply:
added assert